### DISCUSSION TITLE
"On Site" Verification to earn Stars and for "source=survey"

### DISCUSSION THREAD

**Discussion by ElbertCode** [streetcomplete/StreetComplete#4930] [Ideas]
## Problem
There are some feature requests, regarding the further gamification (such as #4561 or #3706), whose implementation is, understandably, rejected, as they could lead to worse data being contributed.

## Possible Solution
Only if a quest is made **on site**, you earn a star for it (which are used for the rank). Other contributions would still be uploaded to OSM, but since they don't matter for the leaderboard, gamification wouldn't be a motivation for cheating.
Obviously this solution doesn't eliminate all possible ways of cheating, e. g. you can still select the surface of a long road without checking if it's the same at the other end but you couldn't just sit at home and answer a few quests to get a better rank.
### Implementation
The app already has a warning displayed if you want to make changes not on site. So the hardest part should already be done, the only thing left is not to count such changes towards the stars and therefore not for the ranking.

**Comment by westnordost** [comment]
Actually, checking if things are (still) the same they have been mapped before I find even more important than adding new data. (As wrong data is worse than no data.) As I understand, your suggestion would lead to users being punished for confirming on-site that the situation _didn't_ change. This doesn't sound reasonable.

**Reply by ElbertCode** [reply]
I think you misunderstood me, my idea was just that contributions should only count for your rank if they were made on site to prevent cheating for the purpose of getting a higher rank. It has nothing to do with whether a contribution adds new data or reviews old data.

**Reply by Helium314** [reply]
Actually I read this as "only increase star count if quest was definitely answered on site" (i.e. using the distance check that's also used for are-you-really-on-site dialog).

@ElbertCode
Could you clarify?
Do you indeed want to require a change to be made (as opposed to just confirming current situation)?

**Reply by mnalis** [reply]
> Actually, checking if things are (still) the same they have been mapped before I find even more important than adding new data. (As wrong data is worse than no data.) 

So do I, but I do not think @ElbertCode suggested otherwise?

> As I understand, your suggestion would lead to users being punished for confirming on-site that the situation didn't change. This doesn't sound reasonable.

I'm not seeing that? The discussion seems to be about exactly opposite - that is, **giving score for confirming on-site** (but **not** giving score when incorrectly using SC by armchair mapping from home) 

How I understood @ElbertCode: the SC should record with each answer whether user was **actually on site** when they answered that quest (meaning GPS was on and GPS location was near quest location. i.e. that they weren't asked  (nor auto-asked/answered) that `Are you sure you checked that on-site?` question that pops up if user is not confirmed to be in vicinity of the quest). IOW, **_if the GPS location/quest location DON'T match, the answer would be recorded normally, but user wouldn't be given any score  for answering it._**

- I would find that reasonable and desirable (perhaps it could be a little less harsh, i.e. users might **alternatively be given lower score** e.g. `0.5` instead of `0`). Reduced score would discourage abusing gamification when not actually on site, while not really impacting regular on-site mappers (who in most cases do not map many kilometers away from GPS location.  Although there are few exceptions - mapping inside closed spaces or in fast moving vehicle might mark many answers as out-of-GPS radius, most mappers probably do not such mapping **exclusively**, so would only be partially affected).

- Additional advantage would be that we could **properly group changesets into ones** which deserve `source=survey` changeset tag, and those which don't -- see https://github.com/streetcomplete/StreetComplete/issues/298 and many issues linked into it.

But implementing it would require some additional work for saving with each quest how far it was from actual GPS location (i.e. does it qualify for `source=survey` changeset tag and full score) -- as many users do not have auto-upload enabled, and if not recorded that information would be lost.

**Reply by ElbertCode** [reply]
My intention was not to separate quests, except between "on site"/"not on site".

And then it is for **on site**:

- the change is uploaded to OSM 
- you get a star in StreetComplete (on which, as far as I understand, the rank is based).

And for changes that were made **not on site**: 

- the change is uploaded to OSM and that's it.

My idea was, that this could prevent cheating by not actually visiting quests, since they would't count for your rank anyway.

**Reply by mnalis** [reply]
@ElbertCode I agree with you. As I noted before (and in issues links in #298), score should only be given (and also `source=survey` tag added) if the user was confirmed to be on the location. 

In fact, if `source=survey` tagging is _fixed_ (currently we believe what user says instead of verifying it by GPS; which is problematic as it might be influenced by gamification as you note) so it is only added in **GPS-confirmed cases**, the scoring change would be trivial (i.e. we would only count StreetComplete changesets tagged with `source=survey`, instead of all of them).
